### PR TITLE
Migrate from Travis-CI to GitHub actions

### PR DESCRIPTION
**Pull request by mejackreed**
With the OSS Travis support changes, migrate to GitHub actions

Recent TravisCI build times:

https://travis-ci.com/github/projectblacklight/spotlight/builds/193901293

Screen Shot 2020-12-09 at 7 44 37 AM

While these have bundler/test app caching built in, they different builds will run serially. The GitHub actions update are run in parallel, having a much faster overall CI completion.

GitHub actions is running about 17m total time. An improvement overall from Travis-CI

### CODE REVIEW

**Review by jkeck — COMMENTED** [review]
Want to fix up the badge in the README too?

File: blacklight-spotlight.gemspec [hunk 1]
```diff
@@ -24,7 +24,7 @@ these collections.)
   s.add_dependency 'acts-as-taggable-on', '>= 5.0', '< 7'
   s.add_dependency 'almond-rails', '>= 0.3.0'
   s.add_dependency 'autoprefixer-rails'
-  s.add_dependency 'blacklight', '~> 7.0'
+  s.add_dependency 'blacklight', '~> 7.12.1'
```
Comment (mejackreed) [inline blacklight-spotlight.gemspec:27]: Seeing an issue with:

```
Spotlight::Search#search_params maps the search to the appropriate facet values
     Failure/Error: expect(subject.search_params.to_hash).to include 'fq' => array_including('{!term f=genre_sim}map')

       expected {"f.corporate_name_ssm.facet.limit" => 11, "f.exhibit_exhibit-title-910_tags_ssim.facet.limit" => 11, "f....blic_bsi:false"], "hl.fl" => [], "qt" => "search", "rows" => 10, "sort" => "score desc, sort_title_ssi asc"} to include {"fq" => #<RSpec::Mocks::ArgumentMatchers::ArrayIncludingMatcher:0x000055b554eb0910 @expected=["{!term f=genre_sim}map"]>}
       Diff:
       @@ -1,14 +1,27 @@
       -"fq" => array_including({!term f=genre_sim}map),
       +"f.corporate_name_ssm.facet.limit" => 11,
       +"f.exhibit_exhibit-title-910_tags_ssim.facet.limit" => 11,
       +"f.genre_ssim.facet.limit" => 11,
       +"f.personal_name_ssm.facet.limit" => 11,
       +"facet" => false,
       +"facet.field" => ["genre_ssim", "personal_name_ssm", "corporate_name_ssm", "subject_geographic_ssim", "subject_temporal_ssim", "language_ssim", "exhibit_exhibit-title-910_tags_ssim"],
       +"facet.pivot" => [],
       +"facet.query" => [],
       +"fl" => "*",
       +"fq" => ["-exhibit_exhibit-title-910_public_bsi:false"],
       +"hl.fl" => [],
       +"qt" => "search",
       +"rows" => 10,
       +"sort" => "score desc, sort_title_ssi asc",
     # ./spec/models/spotlight/search_spec.rb:57:in `block (3 levels) in <top (required)>'
```

Comment (cbeer) [reply blacklight-spotlight.gemspec:27]: I was going to say, I thought I fixed this already.. but I guess code you don't push up doesn't count (maybe I was going to think harder to see if it was a backwards compatibility concern?)

Basically -- change `genre_sim` (which doesn't exist and was probably a typo) to `genre_ssim`

Comment (mejackreed) [reply blacklight-spotlight.gemspec]: Got it thanks!

**Comment by mejackreed** [comment]
I'm now seeing:

```
Failures:

  1) Spotlight::Search for a search matching all items has items
     Failure/Error: expect(subject.documents.size).to eq 55

       expected: 55
            got: 56

       (compared using ==)
     # ./spec/models/spotlight/search_spec.rb:20:in `block (3 levels) in <top (required)>'
```

But locally I'm unable to recreate this using Ruby 2.6 and 2.7. Any guesses?

**Pull request merged**